src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,29 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws RegistrationException {
+        if (user == null) {
+            throw new RegistrationException("User is null");
+        }
+        if (storageDao.get(user.getLogin()) != null
+                && !user.getLogin().equals(storageDao.get(user.getLogin()))) {
+            throw new RegistrationException("User with this login already exists ");
+        }
+        if (user.getLogin().length() < 6 || user.getPassword().length() < 6) {
```

Comment:
will throw NPE here trying to call .length on null. Extract null check for login and password as separate blocks